### DISCUSSION TITLE
more than just frontmatter from pageContext

### DISCUSSION THREAD

**Discussion by cysabi** [mdx-js/mdx#1408] [Q&A] [answered]
Hey, I'd like to access the excerpt of the post from inside of the a template file.

```jsx
const Template = ({ pageContext, children }) => {
    console.log(pageContext.excerpt)
}
```

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @LeptoFlare! :wave: 
Can you expand on your question a bit? In particular:
* How are you using MDX? (Through Gatsby, Next, Webpack, etc)
* What are you using to generate an excerpt? (that isn't a part of MDX itself as far as I'm aware, it's most likely coming from the platform you are using, similar to the question above)
* What is your goal in pulling except into a template?

**Reply by cysabi** [reply]
Hey! I'm using Gatsby, sorry I didn't specify that.

**Reply by ChristianMurphy** [reply]
No problem, thanks for clarifying.
And the other questions?
* What are you using to generate an excerpt?
* What is your goal in pulling except into a template?

**Reply by cysabi** [reply]
`gatsby-plugin-mdx` has a field in the graphql api that returns an excerpt based on the body.
![image](https://user-images.githubusercontent.com/47790183/102812523-241cfe00-4395-11eb-9e10-ea63e015326a.png)

I've attempted to create a page query but this query doesn't seem to be accessible in the props:
```jsx
export const query = graphql`
  query MDXQuery($id: String!) {
    mdx(id: { eq: $id }) {
      id
      body
    }
  }
`
```
I pulled this from the documentation listed here: https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx/?=MDX#mdxrenderer

The goal is to use the excerpt as a description in the metatags.

**Reply by ChristianMurphy** [reply]
A couple things:
1. MDX is the document language itself, the GraphQL mappings is all `gatsby-plugin-mdx` which is managed by the Gatsby team. For Gatsby GraphQL questions, the Gatsby forum (https://github.com/gatsbyjs/gatsby/discussions) may be able to provide more specifics than the MDX one.
2. In your first example, you include `excerpt` as a field, in the second you don't include `excerpt` as a field. The Gatsby docs indicate `excerpt` should be availible https://github.com/gatsbyjs/gatsby/blob/18de9ef795e86fc6cfa0b948cf51157eaca2bff2/docs/docs/how-to/routing/migrate-remark-to-mdx.md#update-usage-in-pages, try adding `excerpt` to the page query

**Reply by cysabi** [reply]
For 2, the *entire* query data is not available in the page component. This includes the `id` and the `body`, The second example is not what's being used in my code, the first one is.

For 1. I see, I'm alright with closing this discussion then

**Reply by ChristianMurphy** [reply]
> The second example is not what's being used in my code, the first one is.

:confused: The first one as in:

> ![image](https://user-images.githubusercontent.com/47790183/102812523-241cfe00-4395-11eb-9e10-ea63e015326a.png)

?
The `excerpt` is right there in the result... :confused: 
I'm really unclear what you're asking, both the first and second example you've shared should work.

**Reply by cysabi** [reply]
Right, it works in the graphql explorer, but it doesn't work in the code itself. which is the `export const query...`
Since this was the wrong place to ask, I'm going to mark as answer

**Reply by ChristianMurphy** [reply]
> I'm alright with closing this discussion then

Either way works, I'm just letting you know that I am not currently a maintainer or user of `gatsby-plugin-mdx`, and my answers are going off publicly available docs.
I'm happy to help, but may not be as insightful as the fine folks at Gatsby who do maintain the plugin.
